### PR TITLE
Wizard recipe: GDB-v10.1.0

### PR DESCRIPTION
**Pull request by Keno**
This pull request contains a new build recipe I built using the BinaryBuilder.jl wizard:

* Package name: GDB
* Version: v10.1.0

@staticfloat please review and merge.

### CODE REVIEW

**Comment by Keno** [comment]
I'm gonna be risky and try adding back `-j${nproc}`. I have a bit of a suspicion that GDB is susceptible to race conditions, since I saw something like that during the build process, but let's find out.

File: G/GDB/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = [
+    Platform("i686", "linux"; libc = "glibc"),
+    Platform("x86_64", "linux"; libc = "glibc"),
+    Platform("aarch64", "linux"; libc = "glibc"),
+    Platform("armv7l", "linux"; call_abi = "eabihf", libc = "glibc"),
+    Platform("armv7l", "linux"; call_abi = "eabihf", libc = "musl"),
```
Comment (giordano) [inline G/GDB/build_tarballs.jl:29]: Maybe it'll work for other Musl platforms?

File: G/GDB/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    Platform("i686", "windows"; ),
+    Platform("x86_64", "windows"; )
+]
+
```
Comment (giordano) [inline G/GDB/build_tarballs.jl:33]:
```suggestion
platforms = expand_cxxstring_abis(platforms)
```

File: G/GDB/build_tarballs.jl (hunk 1 above)
Comment (Keno) [reply G/GDB/build_tarballs.jl:29]: no, it doesn't - I tried them all.

Comment (giordano) [reply G/GDB/build_tarballs.jl:29]: Interesting :confused:

File: G/GDB/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+# The products that we will ensure are always built
+products = [
+    ExecutableProduct("gdbserver", :gdbserver),
+    ExecutableProduct("gdb", :gdb)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    Dependency(PackageSpec(name="GMP_jll", uuid="781609d7-10c4-51f6-84f2-b8444358ff6d"))
```
Comment (giordano) [inline G/GDB/build_tarballs.jl:43]: If we build with v6.1.2, it'll be compatible with older versions of the library
```suggestion
    Dependency(PackageSpec(name="GMP_jll", uuid="781609d7-10c4-51f6-84f2-b8444358ff6d"), v"6.1.2")
```

File: G/GDB/build_tarballs.jl (hunk 1 above)
Comment (giordano) [reply G/GDB/build_tarballs.jl:29]: There are some patches in Alpine Linux: https://git.alpinelinux.org/aports/tree/main/gdb?h=master.

**Pull request merged**